x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/knowledge_base/index.ts

```diff
@@ -235,12 +248,14 @@ export class AIAssistantKnowledgeBaseDataClient extends AIAssistantDataClient {
       }
 
       this.options.logger.debug(`Checking if Knowledge Base docs have been loaded...`);
-      const kbDocsLoaded = (await esStore.similaritySearch(ESQL_DOCS_LOADED_QUERY)).length > 0;
-      if (!kbDocsLoaded) {
-        this.options.logger.debug(`Loading KB docs...`);
-        await loadESQL(esStore, this.options.logger);
-      } else {
-        this.options.logger.debug(`Knowledge Base docs already loaded!`);
+      if (installEsqlDocs) {
```

Comment:
I've added the option to _not_ install the ESQL KB docs to make things easy in case we end up adopting https://github.com/elastic/kibana/pull/192042 for `8.16`.